src/sentry/integrations/vercel/integration.py

```diff
@@ -108,6 +108,75 @@ def get_organization_config(self):
 
         return fields
 
+    def update_organization_config(self, data):
+        # data = {"project_mappings": [[sentry_project_id, vercel_project_id]]}
+
+        metadata = self.model.metadata
+        vercel_client = VercelClient(metadata["access_token"], metadata.get("team_id"))
+        config = self.org_integration.config
+        [sentry_project_id, vercel_project_id] = data["project_mappings"][
+            -1
+        ]  # TODO: update this to work in the case where a project is removed
+        sentry_project = Project.objects.get(id=sentry_project_id)
+        enabled_dsn = ProjectKey.get_default(project=sentry_project)
+        if not enabled_dsn:
+            raise IntegrationError("You must have an enabled DSN to continue!")
+        sentry_project_dsn = enabled_dsn.get_dsn(public=True)
+
+        org_secret = self.create_secret(
+            vercel_client, vercel_project_id, "SENTRY_ORG", sentry_project.organization.slug
+        )
+        project_secret = self.create_secret(
+            vercel_client,
+            vercel_project_id,
+            "SENTRY_PROJECT_%s" % sentry_project_id,
+            sentry_project.slug,
+        )
+        dsn_secret = self.create_secret(
+            vercel_client,
+            vercel_project_id,
+            "NEXT_PUBLIC_SENTRY_DSN_%s" % sentry_project_id,
+            sentry_project_dsn,
+        )
+
+        self.create_env_var(vercel_client, vercel_project_id, "SENTRY_ORG", org_secret)
+        self.create_env_var(vercel_client, vercel_project_id, "SENTRY_PROJECT", project_secret)
+        self.create_env_var(vercel_client, vercel_project_id, "NEXT_PUBLIC_SENTRY_DSN", dsn_secret)
```

Comment:
If this set of requests fails part way through is there a way for the user to restart this process or will it be retried in a celery task?

Reply:
@markstory This is triggered by a button click so the user can just press the "add" button again.

Reply:
:+1: Will the UI let them know the operation failed and tell them to retry?

Reply:
It doesn't yet but I assume we'll add this in the front end similar to how we'll show a toast if the DSN is disabled/non-existent, yeah @scefali ?

Reply:
All it says is "Failed to save". I honestly think that is OK as the natural inclination will be to try again. I don't think it's worth investing in a solution to figure out which errors can be re-tried and showing a different UI experience. That being said, if users start complaining about this, we can add it. But I don't expect retryable errors to be very common.